models/utils.go

```diff
@@ -0,0 +1,286 @@
+package models
```

Comment:
This looks like a very good list of utils, but do we need all the utils? Maybe we can just keep the ones that we are using and remove the rest so that the code contains minimum jargon and is easy to read?